### ISSUE TITLE
job config defaults not loading in launchpad 

### ISSUE THREAD

**Issue by davidz-1** [dagster-io/dagster#6522]
## Summary
Setup:
- single graph definition `graph_1` 
- creating multiple jobs from it `job_pro` and  `job_dev`
- both jobs go to separate repos: 
  - `job_pro`  -> `pro_repo`
  - `job_dev`  -> `dev_repo` 

I like to provide a default configuration for those jobs. Problem is that the launchpad config shows only for `job_pro` on dagit and doesn't load for `job_dev`.
I figured out that if both job definitions go to the same `repository` the config is loaded for both jobs as expected, as soon as they go to different repos the default config is only loaded on `job_pro`.

example:
```python
from my_graph import graph_1

default_config = {
    "ops": {
        "some_op": {
            "config": {
                "nice_key": "cool"
            }
        }
    }
}

job_pro = graph_1.to_job(
    name="job_pro",
    config=default_config
)

job_dev = graph_1.to_job(
    name="job_dev",
    config=default_config
)
```

## Reproduction
Tested this on `0.13.17` and `0.13.18`, both with similar results.

## Additional Info about Your Environment
Tested all local, with spawning `dagit` and accessing a local `workspace.yaml`.

```yaml
---
load_from:
    - python_file:
        relative_path: repositories/pro.py
        working_directory: ./
    - python_file:
        relative_path: repositories/dev.py
        working_directory: ./
```

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

**Comment by alangenfeld** [comment]
Having trouble reproducing this locally. 

Does this repro for you in an incognito window or after clearing local storage?

**Comment by alangenfeld** [comment]
Opening new tabs in the launchpad is probably the best workaround while we work on fixing this

**Comment by hellendag** [comment]
I'll try some stuff to see what works.

**Comment by hellendag** [comment]
I believe this is resolved by #7559.

There is an edge case noted by @alangenfeld above, in which the launchpad editor has been intentionally cleared of content. We will save this config session, as it may be what you see the next time you load the launchpad. This should be uncommon, since having a default config implies that you probably won't clear the editor and end up in that state.

If you do find yourself in this edge case, creating a new config tab should correctly load the default config, and you can delete the old empty tab.

You should see a correctly populated config tab in the launchpad when:

- Viewing a job where you have never edited config
- Viewing a job after clearing localStorage
- Creating a new config tab

This change will be in the upcoming release. Please let us know if you continue running into this problem after that!
